protocol/bc/asset_witness.go

```diff
@@ -0,0 +1,76 @@
+package bc
+
+import (
+	"fmt"
+	"io"
+
+	"chain/crypto/sha3pool"
+	"chain/encoding/blockchain"
+	"chain/errors"
+)
+
+type AssetWitness struct {
```

Comment:
For the most part, this PR seems to be moving our Go types to more closely match the datatypes described in the spec, but AssetWitness is an exception. What is it?

It would be good to have godoc, at least for newly-introduced types. The ones that *do* map 1-1 can be short, like "Type Foo represents a Foo as defined in Chain Protocol. See https://chain.com/docs/protocol/specifications/data."